docs/architecture/adr-020-protobuf-transaction-encoding.md

```diff
@@ -83,9 +83,33 @@ message SignerInfo {
         // address can be used for accounts that already have a public key in state 
         bytes address = 2;
     }
-    SignMode mode = 3;
+
+    ModeInfo mode_info = 3;
 }
 
+message ModeInfo {
+    oneof sum {
+        Single single = 1;
+        Multi multi = 2;
+    }   
+   
+    // Single is the mode info for a single signer
+    message Single {
+        SignMode mode = 1;
+        // locale may be used in the future with TEXTUAL signing modes
+        string locale = 2;
+    }
+   
+    // Multi is the mode info for a multisig public key
+    message Multi {
+        // bitarray specifies which keys within the multisig are signing
+        CompactBitArray bitarray = 1;
+        // mode_infos is the corresponding modes of the signers of the multisig
+        // which could include nested multisig public keys
```

Comment:
This comment is confusing to me. It reads as if nested multisig public keys a possible signing mode, yet I don't see anything corresponding to "nested multisig public keys" in the mode enum.

Reply:
It wouldn't exist in the mode enum. It's a recursive set of `ModeInfo` messages

Reply:
For the latest design of the actual pub keys see #5997

Reply:
But at the proto level, this would be a flattened list of `ModeInfo` messages right? If I understand correctly, the purpose for including the `ModeInfo` list at all is only for accurate gas cost estimations. 

Or does `repeated` have some support for recursive lists?